src/signingClient.ts

```diff
@@ -682,6 +684,59 @@ export class ISCNSigningClient {
     return response;
   }
 
+  async createGenericGrant(
+    senderAddress: string,
+    granteeAddress: string,
+    messageTypeUrl: string,
+    expirationInMs: number,
+    { fee: inputFee, gasPrice, ...signOptions }: ISCNSignOptions = {},
+  ): Promise<TxRaw | DeliverTxResponse> {
+    const client = this.signingClient;
+    if (!client) throw new Error('SIGNING_CLIENT_NOT_CONNECTED');
+    const messages = [formatMsgGrantGenericAuthorization(
+      senderAddress,
+      granteeAddress,
+      messageTypeUrl,
+      expirationInMs,
+    )];
+    let fee = inputFee;
+    if (!fee) {
+      fee = {
+        amount: [{
+          amount: new BigNumber(GRANT_SEND_AUTH_GAS)
+            .multipliedBy(gasPrice || DEFAULT_GAS_PRICE_NUMBER).toFixed(0, 0),
+          denom: this.denom,
+        }],
+        gas: GRANT_SEND_AUTH_GAS.toString(),
+      };
+    } else if (gasPrice) {
+      throw new Error('CANNOT_SET_BOTH_FEE_AND_GASPRICE');
+    }
+    const response = await signOrBroadcast(senderAddress, messages, fee, client, signOptions);
+    return response;
+  }
+
+  async revokeGenericGrant(
+    senderAddress: string,
+    granteeAddress: string,
+    messageTypeUrl: string,
+    { fee: inputFee, gasPrice, ...signOptions }: ISCNSignOptions = {},
+  ): Promise<TxRaw | DeliverTxResponse> {
+    const client = this.signingClient;
+    if (!client) throw new Error('SIGNING_CLIENT_NOT_CONNECTED');
+    const messages = [formatMsgRevokeGenericAuthorization(
+      senderAddress, granteeAddress, messageTypeUrl,
+    )];
+    if (inputFee && gasPrice) throw new Error('CANNOT_SET_BOTH_FEE_AND_GASPRICE');
+    const fee = inputFee || formatGasFee({
+      gas: REVOKE_SEND_AUTH_GAS,
+      gasPrice,
+      denom: this.denom,
+    });
+    const response = await signOrBroadcast(senderAddress, messages, fee, client, signOptions);
+    return response;
+  }
+
   async createSendGrant(
```

Comment:
Is it preferred to refactor `createSendGrant` with `createGenericGrant` too, or just let it be?

Reply:
createGenericGrant grants a `GenericAuthorization` which accept `msg` as param
createSendGrant grants a `SpendLimitAuthorization` which accept `spendLimit` as param
so we would need further refactoring if we want the create function to be reusable
on the other hand, all revoke functions just take the `msg` url as param